### PR TITLE
Allow customization of API Versions when using Helm Template

### LINKED ISSUE #3377: Add ability to pass Capabilities.APIVersions to template and lint
When using the template and lint commands the Capabilities.APIVersions is an empty set because it uses defaults which are currently an empty set. Might be good to provide a set or provide the capability to pass in a set.

Note, lint doesn't let you pass in the version of k8s either. That may also be useful.

### PR DESCRIPTION
**Pull request by bnookala**
Signed-off-by: Bhargav Nookala <nooknb@gmail.com>

**What this PR does / why we need it**:

Resolves https://github.com/helm/helm/issues/3377 for `helm template`. 
This PR adds `--api-version` (or `-a`) to add custom API versions to Capabilites.APIVersion, allowing a user to correctly render templates that depend on certain API versions.

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains unit tests
- [x] this PR contains documentation

### CODE REVIEW

**Comment by bnookala** [comment]
hi @bacongobbler - would you be able to take a look at this addition to `helm template`?

**Comment by kramvan1** [comment]
+1

**Comment by alexmt** [comment]
This would be very useful!

**Comment by basvandenbrink** [comment]
Agree, would be nice if this is going to be merged!

**Comment by sundag** [comment]
Pending on this feature to be available.

**Comment by alexec** [comment]
+1

**Comment by sbose78** [comment]
I like this @bnookala ! 


Could we have this feature in helm3 / "dev-v3" branch as well, please?

File: cmd/helm/template.go [hunk 1]
```diff
@@ -103,6 +104,7 @@ func newTemplateCmd(out io.Writer) *cobra.Command {
 	f.StringArrayVar(&t.fileValues, "set-file", []string{}, "set values from respective files specified via the command line (can specify multiple or separate values with commas: key1=path1,key2=path2)")
 	f.StringVar(&t.nameTemplate, "name-template", "", "specify template used to name the release")
 	f.StringVar(&t.kubeVersion, "kube-version", defaultKubeVersion, "kubernetes version used as Capabilities.KubeVersion.Major/Minor")
+	f.StringArrayVarP(&t.apiVersions, "api-version", "a", []string{}, "kubernetes api versions used for Capabilities.APIVersions")
```
Comment (thomastaylor312) [inline cmd/helm/template.go]: This should probably be plural (`api-versions`)

**Review by thomastaylor312 — CHANGES_REQUESTED** [review]
This is looking great! Just need a super small change on the flag name and this will be good to go

File: cmd/helm/template.go (hunk 1 above)
Comment (bnookala) [reply cmd/helm/template.go]: @thomastaylor312 done!

File: cmd/helm/template_test.go [hunk 2]
```diff
@@ -175,6 +175,13 @@ func TestTemplateCmd(t *testing.T) {
 			expectKey:   "subchart1/templates/service.yaml",
 			expectValue: "kube-version/major: \"1\"\n    kube-version/minor: \"6\"\n    kube-version/gitversion: \"v1.6.0\"",
 		},
+		{
+			name:        "check_kube_api_versions",
+			desc:        "verify --api-version overrides kubernetes api versions",
+			args:        []string{subchart1ChartPath, "--api-version", "helm.k8s.io/test"},
```
Comment (thomastaylor312) [inline cmd/helm/template_test.go]: This flag needs to be updated with the new plural flag

**Comment by bacongobbler** [comment]
might also be worth to rebase against master once more now that #6577 has been merged. It includes a fix where coreutils was bumped underneath us, causing CI to fail :) thanks @bnookala!

File: cmd/helm/template_test.go (hunk 2 above)
Comment (bnookala) [reply cmd/helm/template_test.go]: oops - fixed!

**Review by bacongobbler — APPROVED** [review]
I know it's sorta last minute here, but I would LOVE to see some documentation on how users can consume this flag, where it's useful, etc.

Regardless thank you for the PR! Looking forward to seeing this in 2.15. I'll leave it for @thomastaylor312 to review and merge since he kicked off the review process.

**Comment by bnookala** [comment]
@bacongobbler oh for sure, can totally do that - do you have a suggestion on where the docs should live? at a quick glance, thinking the "thinking about charts" section might be a good option: https://helm.sh/docs/using_helm/#thinking-about-charts

**Comment by thomastaylor312** [comment]
@bnookala I would actually suggest somewhere here: https://helm.sh/docs/chart_template_guide/#debugging-templates

Info on the template command was never added there and would be useful to have

**Review by thomastaylor312 — APPROVED** [review]
Thanks for all the hard work here

**Pull request merged**